tests/di/service_collection_test.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace
+{
+	class phpbb_service_collection_test extends \phpbb_test_case
+	{
+		/**
+		 * @var \phpbb\di\service_collection
+		 */
+		protected $service_collection;
+
+		public function setUp()
+		{
+			// Set up service container
+			$phpbb_root_path = dirname(__FILE__) . '/';
+
+			$filename = $phpbb_root_path . '../tmp/container.php';
+			if (is_file($filename))
+			{
+				unlink($filename);
+			}
+
+			$builder = new phpbb_mock_phpbb_di_container_builder($phpbb_root_path . 'fixtures/', 'php');
+			$builder->with_config_path($phpbb_root_path . 'fixtures/collection_config/');
+			$builder->without_cache();
+			$container = $builder->get_container();
+			$this->service_collection = $container->get('service_collection');
+
+			parent::setUp();
+		}
+
+		public function test_service_collection_get_service_names()
+		{
+			global $service_initialized;
+			$service_initialized = false;
+
+			$service_names = $this->service_collection->get_service_names();
+			$this->assertTrue(in_array('collection_service1', $service_names));
```

Comment:
First of all, there is assertContains. But then, this should also rather be assertSame with an array if the order is known. Otherwise $service_names can contain more elements.